[event-hubs] Removed checkpointstore sample and reference it instead

willmtemple opened the pull request:
The checkpointstore sample in event-hubs introduces a circular package dependency in order to be built. This sample already exists in TS in the checkpointstore package, so remove it from this package and reference that instead. I included a comment in receiveEvents.js and receiveEvents.ts pointing to the other sample on GitHub.

I also converted the TS sample in checkpointstore to have a CommonJS-compatible copy as well, which is referenced in the TS version of the receiveEvents sample.

ramya-rao-a commented:
Make a check that there is no stale link in the package readme for event hubs to the checkpoint store sample

willmtemple commented:
I removed all the dead links in the README to the outdated samples as well as the link definitions at the bottom. The only place "checkpoint" appears in the samples directory is in the link to the checkpointstore repo that I added to the receiveEvents files.

richardpark-msft commented on sdk/eventhub/event-hubs/samples/javascript/README.md:
```diff
@@ -18,8 +18,7 @@ These sample programs show how to use the JavaScript client libraries for Azure
 | [usingAadAuth.js][usingaadauth]                                           | demonstrates how to instantiate EventHubsClient using AAD token credentials obtained from using Service Principal Secrets                                                               |
-| [receiveEventsUsingCheckpointStore.js][receiveeventsusingcheckpointstore] | demonstrates how to use the EventHubConsumerClient to process events from all partitions of a consumer group in an Event Hubs instance (requires running the `sendEvents` sample first) |
-| [receiveEvents.js][receiveevents]                                         | demonstrates how to use the EventHubConsumerClient to process events from all partitions of a consumer group in an Event Hubs instance, as well as checkpointing along the way.         |
+| [receiveEvents.js][receiveevents]                                         | demonstrates how to use the EventHubConsumerClient to process events from all partitions of a consumer group in an Event Hubs instance.                                                 |
```
Is there any reason we can't leave the link in and just point to the version in the eventhub-checkpointstore-blob package's sample folder? It would make it more easily discoverable.

ramya-rao-a replied:
Thats a great idea

willmtemple replied:
I added a link and a short explanation below the table. All throughout the samples everything else in these tables has the property of "anything in the table is in the zip file," and I didn't want to break that consistency so I made it a little more visually distinct. I think we are good to merge here.

ramya-rao-a commented on sdk/eventhub/event-hubs/samples/javascript/README.md:
```diff
@@ -67,9 +68,9 @@ Take a look at our [API Documentation][apiref] for more information about the AP
-[receiveeventsusingcheckpointstore]: https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/eventhub/event-hubs/samples/javascript/receiveEventsUsingCheckpointStore.js
 [receiveevents]: https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/eventhub/event-hubs/samples/javascript/receiveEvents.js
 [apiref]: https://docs.microsoft.com/javascript/api/@azure/event-hubs
+[checkpointing]: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/eventhubs-checkpointstore-blob/samples/receiveEventsUsingCheckpointStore.js
```
This is going to a 404

willmtemple replied:
The file that link points to is in this commit, so it will not 404 once it is merged.

ramya-rao-a commented on sdk/eventhub/event-hubs/samples/javascript/README.md:
```diff
@@ -18,8 +18,9 @@ These sample programs show how to use the JavaScript client libraries for Azure
-| [receiveEvents.js][receiveevents]                                         | demonstrates how to use the EventHubConsumerClient to process events from all partitions of a consumer group in an Event Hubs instance, as well as checkpointing along the way.         |
+| [receiveEvents.js][receiveevents]                                         | demonstrates how to use the EventHubConsumerClient to process events from all partitions of a consumer group in an Event Hubs instance.                                                 |
+
+**Note**: `EventHubConsumerClient` supports checkpointing using the `@azure/eventhubs-checkpointstore-blob` and `@azure/storage-blob` packages. For an example that demonstrates how to use this functionality, see [this link][checkpointing].
```
Link texts should have contextual information to help screenreaders

```suggestion
**Note**: `EventHubConsumerClient` supports checkpointing using the `@azure/eventhubs-checkpointstore-blob` and `@azure/storage-blob` packages. For an example that demonstrates how to use this functionality, see the sample [receiveEventsUsingCheckpointStore.js][checkpointing].
```

willmtemple replied:
Addressed this in the latest commit
